cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -0,0 +1,379 @@
+	resourceName, groupId string, artifactId string, extraCondition string) {
+	insertedString := ""
+	extraCondition = strings.TrimSpace(extraCondition)
+	if extraCondition != "" {
+		insertedString = " and " + extraCondition
+	}
+	log.Printf("Detected '%s' because found dependency '%s:%s' in pom.xml file%s.",
+		resourceName, groupId, artifactId, insertedString)
+}
+
+func detectServiceBus(azdProject *Project, springBootProject *SpringBootProject) {
+	// we need to figure out multiple projects are using the same service bus
+	detectServiceBusAccordingToJMSMavenDependency(azdProject, springBootProject)
+	detectServiceBusAccordingToSpringCloudStreamBinderMavenDependency(azdProject, springBootProject)
+}
+
+func detectServiceBusAccordingToJMSMavenDependency(azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-starter-servicebus-jms"
+	for _, projectDependency := range springBootProject.mavenProject.Dependencies {
+		if projectDependency.GroupId == targetGroupId && projectDependency.ArtifactId == targetArtifactId {
+			newDependency := AzureDepServiceBus{
+				IsJms: true,
+			}
+			azdProject.AzureDeps = append(azdProject.AzureDeps, newDependency)
+			logServiceAddedAccordingToMavenDependency(newDependency.ResourceDisplay(), targetGroupId, targetArtifactId)
+			break
+		}
+	}
+}
+
+func detectServiceBusAccordingToSpringCloudStreamBinderMavenDependency(
+	azdProject *Project, springBootProject *SpringBootProject) {
+	var targetGroupId = "com.azure.spring"
+	var targetArtifactId = "spring-cloud-azure-stream-binder-servicebus"
+	for _, projectDependency := range springBootProject.mavenProject.Dependencies {
+		if projectDependency.GroupId == targetGroupId && projectDependency.ArtifactId == targetArtifactId {
+			bindingDestinations := findBindingDestinations(springBootProject.applicationProperties)
+			destinations := make([]string, 0, len(bindingDestinations))
+			for _, destination := range bindingDestinations {
+				if !contains(destinations, destination) {
+					destinations = append(destinations, destination)
+				}
+			}
+			newDep := AzureDepServiceBus{
+				Queues: destinations,
+				IsJms:  false,
+			}
+			azdProject.AzureDeps = append(azdProject.AzureDeps, newDep)
+			logServiceAddedAccordingToMavenDependency(newDep.ResourceDisplay(), targetGroupId, targetArtifactId)
+			for bindingName, destination := range bindingDestinations {
+				log.Printf("  Detected Service Bus queue [%s] for binding [%s] by analyzing property file.",
+					destination, bindingName)
+			}
+			break
+		}
+	}
+}
+
+func detectEventHubs(azdProject *Project, springBootProject *SpringBootProject) {
+	// we need to figure out multiple projects are using the same event hub
+	detectEventHubsAccordingToSpringCloudStreamBinderMavenDependency(azdProject, springBootProject)
+	detectEventHubsAccordingToSpringCloudStreamKafkaMavenDependency(azdProject, springBootProject)
```

Comment:
Same here: One function for one detection rule, this will make it easier to add new rule.